a10_octavia/common/utils.py

```diff
@@ -36,6 +36,32 @@ def validate_ipv4(address):
         raise ConfigFileValueError('Invalid IPAddress value given in configuration: ' + address)
 
 
+def patch_ipv4_suffix(suffix, subnet):
+    """Patch the partial IPv4 suffix with subnet provided
+       If suffix contains valid ip returns suffix itself.
+       Returns patched_ip if valid patch else returns None
+    """
+    suffix_octets = suffix.split('.')
+    if suffix_octets[0] == '':
+        suffix_octets.pop(0)
+
+    suffix_len = len(suffix_octets)
+    if not 0 < suffix_len <= 4:
+        return None
+
+    prefix_octets = subnet.split('.')
```

Comment:
We can't do this with the subnet at the config stage since we don't know the subnet or the netmask. We are going to handle the subnet side of things later.

Reply:
At the config level we provide a fixed string like "10.0.0.0" like I specified in Technical Approach point 1. I just named the string subnet and we use the prefix only. I can rename the argument to prefix.
At the runtime when we get the cidr subnet, we need to pass that as prefix to the function.

Reply:
We don't want to have to have them specify the exact subnet ip in the config file. That defeats the whole purpose of this feature. They could easily create the LB which will cause this information to be stored long before they decide where they want the members to be created.

The runtime check doesn't really get us anywhere. I could pass `15.8.9` and subnet `12.13.0.0` and it'll spit out `12.15.8.9` but the netmask is `255.255.240.0` it's invalid so we just did a bunch of work for nothing. Better to just set it to `0.15.8.9`. Then we can just confirm it's in range and do a bitwise & to set the network ip. This is faster and less LOC than what is being done here.

All this code should do is transform `5`, `.5`, `.5.2.1` to `0.0.0.5`, `0.0.0.5`, `0.5.2.1`. Then check that is is an actually valid IP so we don't have a weird `.777` being passed in or something when they meant `.77`

Reply:
Did the changes.